tests/core/policies/test_rule_policy.py

```diff
@@ -922,6 +920,113 @@ async def test_predict_form_action_if_multiple_turns():
     assert_predicted_action(prediction, domain, form_name, is_no_user_prediction=True)
 
 
+async def test_predict_slot_initial_value_not_required_in_rule():
+    domain = Domain.from_yaml(
+        """
+intents:
+- i1
+actions:
+- action1
+slots:
+  s_cat1:
+    type: categorical
+    values:
+      - v1
+      - v2
+    initial_value: v1
+"""
+    )
+
+    rule = DialogueStateTracker.from_events(
+        "slot rule",
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": "i1"}),
+            ActionExecuted("action1"),
+            SlotSet("s_cat1", "v2"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+        domain=domain,
+        slots=domain.slots,
+    )
+    rule.is_rule_tracker = True
+
+    policy = RulePolicy()
+    policy.train([rule], domain, RegexInterpreter())
+
+    form_conversation = DialogueStateTracker.from_events(
+        "slot rule test",
+        evts=[
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": "i1"}),
```

Comment:
Are these two events needed? I think they can be removed...

Reply:
the `UserUttered` and the `action1`? They are not needed to test this fix, that is true. I was actually modelling it on your example in the issue, but maybe better to have it more succinct.

Reply:
Yep, I meant those two. I know, my example wasn't the best, but let's remove them from this test to not confuse anyone else with unnecessary stuff.

Reply:
ok I've reduced it to what I think is the smallest real example.